File: Programmers/Lv_1/체육복.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    sort(lost.begin(),lost.end());
    sort(reserve.begin(),reserve.end());
    
    for (int i = 0; i < lost.size(); i++)
    {
        for (int j = 0; j < reserve.size(); j++)
        {
            if (lost[i] == reserve[j])
            {
                lost.erase(lost.begin()+i);
                reserve.erase(reserve.begin()+j);
                i--; j--;
            }
        }
    }
    
    for (int i = 0; i < lost.size(); i++)
    {
        for (int j = 0; j < reserve.size(); j++)
        {
            if (lost[i]+1 == reserve[j] || lost[i]-1 == reserve[j])
            {
                lost.erase(lost.begin()+i);
                reserve.erase(reserve.begin()+j);
                i--; j--;
                break;
            }
        }

    }
    
    answer = n-lost.size();
    return answer;
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int n, lostN, reserveN, data;
    vector<int> lost, reserve;
    cin >> n >> lostN;
    for (int i = 0; i < lostN; i++) {
        cin >> data;
        lost.push_back(data);
    }
    cin >> reserveN;
    for (int i = 0; i < reserveN; i++) {
        cin >> data;
        reserve.push_back(data);
    }

    cout << solution(n, lost, reserve);

    return 0;
}

File: Programmers/Lv_1/체육복.py
def solution(n, lost, reserve):
    _reserve = [r for r in reserve if r not in lost]
    _lost = [l for l in lost if l not in reserve]
    
    for r in _reserve:
        f = r - 1
        b = r + 1
        
        if f in _lost:
            _lost.remove(f)
        elif b in _lost:
            _lost.remove(b)
            
    return n - len(_lost)

def solution2(n, lost, reserve):
    answer = 0
    
    for i in range(1, n+1):
        if i not in lost:
            answer += 1
        else:
            if i in reserve:
                answer += 1
                reserve.remove(i)
                lost.remove(i)
                
    for i in lost:
        if i-1 in reserve:
            answer += 1
            reserve.remove(i-1)
            
        elif i+1 in reserve:
            answer += 1
            reserve.remove(i+1)
            
    return answer        
            